plugins/urls/__init__.py

```diff
@@ -178,113 +219,139 @@ def message_handler(self, event=MessageReceived):
-                        self.logger.exception(_("Error catching URL"))
-
-                title, domain = self.parse_title(url)
-
-                self.logger.trace(_("Title: %s") % title)
-
-                if isinstance(target, Channel):
-                    if protocol.name not in self.channels:
-                        with self.channels:
-                            self.channels[protocol.name] = {
-                                target.name: {"last": url,
-                                              "status": "on",
-                                              "shortener":
-                                              "tinyurl"}
-                            }
-                    if target.name not in self.channels[protocol.name]:
-                        with self.channels:
-                            self.channels[protocol.name][target.name] = {
-                                "last": url,
-                                "status": "on",
-                                "shortener":
-                                "tinyurl"
-                            }
-                    else:
-                        with self.channels:
-                            self.channels[protocol.name][target.name]["last"] \
-                                = url
-                    if title is None:
-                        return
+            if _url is None:
+                continue
 
-                    if domain is not None and "/" in domain:
-                        domain = domain.split("/")[0]
-                    if domain is None:
-                        target.respond(title)
-                    else:
-                        target.respond("\"%s\" at %s" % (title, domain))
-                elif isinstance(target, User):
-                    if title is None:
-                        return
+            # Check redirects, following as necessary
+
+            redirects = 0
+            max_redirects = self.config.get("redirects", {}).get("max", 15)
+            domains = self.config.get("redirects", {}).get("domains", [])
+
+            self.logger.debug("Checking redirects...")
+
+            while _url.domain in domains and redirects < max_redirects:
+                redirects += 1
+
+                session = Session()
+
+                #: :type: requests.Response
+                r = yield session.get(str(_url), allow_redirects=False)
+
+                if r.is_redirect:
+                    # This only ever happens when we have a well-formed
+                    # redirect that could have been handled automatically
+
+                    redirect_url = r.headers["location"]
+
+                    self.logger.debug(
+                        "Redirect [{0:03d}] {1}".format(
+                            redirects, redirect_url
+                        )
+                    )
+
+                    _url = self.match_to_url(extract_urls(redirect_url)[0])
+
+            lazy_request = LazyRequest(req_args=[str(_url)])
```

Comment:
Think I've got it, but..

```
[23:23:49] <@g> QAsen, should params being put into sqlite's sql be bytes or unicode?
[23:25:00] <@g> I'm inclined to say bytes because it works but I dunno how sqlite handles that
```